feat(core): Allow InputProvider to use a configurable container (#546)
Workday/canvas-kit: modules/core/react/README.md

import * as React from 'react';
import {InputProvider} from '@workday/canvas-kit-react';

<div>
  <InputProvider />
  {/* All your components containing any Canvas components */}
<div>;
```

> This will result in these event listeners being added to the body as default

Reviewer: ```suggestion
This will result in these data attributes being added to the body element (by default)
```

> Note: detecting intent will add scroll and mouse positioning listeners which could affect
> performance.

##### `container: HTMLElement | React.RefObject<HTMLElement>`

> The containing element in which the InputProvider is rendered into. This property should be set to

> > the element that the application root goes - not containing element of content. Note: If the
> > element is null, it will default to `document.body`.

Reviewer: ```suggestion
> the element that the application root goes - not containing element of content.

Default: `document.body` 
```

> Note: detecting intent will add scroll and mouse positioning listeners which could affect
> performance.

##### `container: HTMLElement | React.RefObject<HTMLElement>`

> > The containing element in which the InputProvider is rendered into. This property should be set to
> > the element that the application root goes - not containing element of content.

Reviewer: ```suggestion
> The containing element in which the InputProvider attaches its data attributes. This property should
> be set to an element that is an ancestor of all your Canvas components.
```
